src/cssls.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+const {
+  Matrix,
+  LuDecomposition,
+  CholeskyDecomposition,
+  solve,
+} = require('ml-matrix');
+
+const sortCollectionSet = require('./util/sortCollectionSet');
+/**
+ * (Combinatorial Subspace Least Squares) - subfunction for the FC-NNLS
+ * @private
+ * @param {Matrix} XtX
+ * @param {Matrix} XtY
+ * @param {Array} Pset
+ * @param {Numbers} l
+ * @param {Numbers} p
+ */
+
+//To do: Another possibility to explore would be to use cholesky factorisation to inverse XtX since is automatically positive semi-definite
+//Look at books at epfl library, could reduce by a half the computation burden
+
+function cssls(XtX, XtY, Pset, l, p) {
+  // Solves the set of equation XtX*K = XtY for the variables in Pset
+  // if XtX (or XtX(vars,vars)) is singular, performs the svd and find pseudoinverse, otherwise (even if ill-conditioned) finds inverse with LU decomposition and solves the set of equation
+  // it is consistent with matlab results for ill-conditioned matrices (at least consistent with test 'ill-conditionned square X rank 2, Y 3x1' in cssls.test)
+
+  let K = Matrix.zeros(l, p);
+  if (Pset === null) {
+    let luXtX = new LuDecomposition(XtX);
+    if (luXtX.isSingular() === false) {
+      K = luXtX.solve(Matrix.eye(l)).mmul(XtY);
+    } else {
+      K = solve(XtX, XtY, { useSVD: true });
+    }
+    /*let choXtX = new CholeskyDecomposition(XtX);
```

Comment:
please remove commented code here and in other places

Reply:
Done